libs/template/file_test.go

```diff
@@ -109,3 +115,37 @@ func TestTemplateCopyFilePersistToDiskForWindows(t *testing.T) {
 	// fs.FileMode values we can use for different operating systems.
 	testCopyFile(t, 0666)
 }
+
+func TestShouldUseImportNotebook(t *testing.T) {
+	ctx := context.Background()
+	data := []byte("# Databricks notebook source\n print('hello')")
+
+	assert.False(t, shouldUseImportNotebook(ctx, "./foo/bar", data))
+	assert.False(t, shouldUseImportNotebook(ctx, "./foo/bar.ipynb", data))
+	assert.False(t, shouldUseImportNotebook(ctx, "/Workspace/foo/bar", data))
+	assert.False(t, shouldUseImportNotebook(ctx, "/Workspace/foo/bar.ipynb", data))
+
+	t.Setenv("DATABRICKS_RUNTIME_VERSION", "14.3")
```

Comment:
This should use `env.Set`.

Reply:
With `env.Set` it adds the variable to the context and passes it along in the context.

If it broke then either 1) you weren't capturing the returned `ctx`, or 2) the function doesn't use `env.Get` to access the environment variables.